### PR TITLE
fix: loosen version constraints

### LINKED ISSUE #44: Tokio Version
The tokio version is being explicitly locked at v1.22.

This is a problem with projects building with solana dependencies, that are locked at v1.14 of tokio.

We forked this repo and specified the tokio dependencies as v1 (not specifying a particular minor version). This fork successfully builds with solana dependencies. Which likely suggests that specifying v1.22 tokio here is unnecessarily strict.

Let's not specify the minor version of tokio here?

### PR DESCRIPTION
**Pull request by chris13524**
# Description

Removes specific version req's. We should be as permissive as possible, letting the downstream app customize the versions. If this library requires a specific minor version or patch, then we can use `^`, `~` or `*` as appropriate

Resolves  #44

## How Has This Been Tested?

Not tested

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: relay_rpc/Cargo.toml [hunk 1]
```diff
@@ -26,32 +26,32 @@ derive_more = { version = "0.99", default-features = false, features = [
...
 ed25519-dalek = { git = "https://github.com/dalek-cryptography/ed25519-dalek.git", rev = "7529d65" }
 rand = "0.7"
 chrono = { version = "0.4", default-features = false, features = [
     "std",
     "clock",
 ] }
-regex = "1.7"
-once_cell = "1.16"
-jsonwebtoken = "8.1"
+regex = "1"
```
Comment (heilhead) [inline relay_rpc/Cargo.toml:39]: So what happens if `relay_rpc` is using features added in `1.7` release, while the consumer code forces e.g. `1.0` version?

**Comment by xDarksome** [comment]
https://users.rust-lang.org/t/psa-please-specify-precise-dependency-versions-in-cargo-toml/71277

**Comment by heilhead** [comment]
Agreed with Nodar and that PSA. I don't think we should make this change, because of how cargo actually resolves dependency versions.

File: relay_rpc/Cargo.toml (hunk 1 above)
Comment (chris13524) [reply relay_rpc/Cargo.toml:39]: If we needed features/fixes from the 1.7 release, then we would set the minimum to 1.7 not 1 as I did

**Comment by chris13524** [comment]
The PSA linked actually doesn't discourage loosening version constraints like I did, but requests that when doing so that `cargo +nightly -Z minimal-versions update` is used in CI. Lose constraints is still good.

I added this to CI, and looks like indeed these are not appropriate version constraints. Furthermore, I tested minimal-versions against the `main` branch and it also fails there. Looks like (many) other crates are also lying about what versions they support.

As discussed in the thread, fixing this requires all downstream crates getting their dependencies right. Hopefully one day we'll get there and we'll be able to install this CI and loosen our own constraints. But closing this until then.

**Pull request closed without merging**